### PR TITLE
Fix syntax highlighting of several preprocessor keywords to not require a trailing space

### LINKED ISSUE #174: (C#) #endregion directive is not being highlighted in the editor
_From @kurozael on April 5, 2016 9:18_

- VSCode Version: 0.10.11
- OS Version: OSX 10.11.4 (El Capitan)

Steps to Reproduce:

1. Open Visual Studio Code in a C# project.
2. Create a #region, #endregion block.

You'll notice that #endregion is not highlighted in the same way that #region is.

noendregionhighlight

_Copied from original issue: Microsoft/vscode#4947_

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #174

Before:

image

After:

image

### CODE REVIEW

**Comment by DustinCampbell** [comment]
@Pilchie, how do you like this one? :smile:

**Comment by Pilchie** [comment]
I hate regular expressions, but appart from that, :+1:

File: syntaxes/csharp.json [hunk 1]
```diff
@@ -553,7 +553,7 @@
           "name": "meta.preprocessor.cs"
         },
         {
-          "match": "^\\s*#\\s*(if|else|elif|endif|define|undef|warning|error|line|pragma|region|endregion)\\b\\s*([^\\/]+)\\s",
+          "match": "^\\s*#\\s*(if|else|elif|endif|define|undef|warning|error|line|pragma|region|endregion)\\b\\s*([^\\/]+)\\s*",
```
Comment (gregg-miskelly) [inline syntaxes/csharp.json]: Could you just remove the '\\s' entirely? Or does the regex need to match the entire line?

**Comment by leonardosnt** [comment]
:+1:

**Pull request closed without merging**